extensions/wikia/PortableInfobox/services/Parser/Nodes/NodeImage.php

```diff
@@ -5,18 +5,22 @@
 
 class NodeImage extends Node {
 	const ALT_TAG_NAME = 'alt';
+	const CAPTION_TAG_NAME = 'caption';
 
 	public function getData() {
 		$title = $this->getImageAsTitleObject( $this->getRawValueWithDefault( $this->xmlNode ) );
 		$ref = null;
 		$alt = $this->getValueWithDefault( $this->xmlNode->{self::ALT_TAG_NAME} );
+		$caption = $this->getValueWithDefault( $this->xmlNode->{self::CAPTION_TAG_NAME} );
 
 		wfRunHooks( 'PortableInfoboxNodeImage::getData', [ $title, &$ref, $alt ] );
 
 		return [
 			'url' => $this->resolveImageUrl( $title ),
-			'name' => ( $title ) ? $title->getDBkey() : '',
+			'name' => ( $title ) ?htmlspecialchars( $title->getText() ) : '',
+			'key' => ( $title ) ? htmlspecialchars( urlencode( $title->getDBKey() ) ) : '',
```

Comment:
And here. Per offline discussion, we should do the alterations just prior to passing this to the template